docs/slsa.md

```diff
@@ -0,0 +1,99 @@
+# SLSA provenance
```

Comment:
Should the file be named `slsa-provenance.md` instead?

Also as discussed, we could replace this section https://github.com/moby/buildkit/blob/master/docs/build-repro.md#build-dependencies and link to this file?

Reply:
I think we can do the build dependencies update as a follow-up? I think the whole page probably needs reworking with the introduction of provenance, since we're reworking how buildinfo works entirely.

Agreed on the `slsa-provenance.md` rename :tada: